crml/nft/src/lib.rs

```diff
@@ -974,6 +1001,130 @@ decl_module! {
 				None => Err(Error::<T>::NotForFixedPriceSale.into()),
 			}
 		}
+
+		/// Create an offer on a token
+		/// Locks funds until offer is accepted, rejected or cancelled
+		#[weight = 0]
```

Comment:
set weights + use `#[transactional]` on all these methods that can fail partway through